### PR TITLE
Organise CSS Z-index

### PR DESCRIPTION
**Pull request by SajidAlamQB**
Related to: https://github.com/kedro-org/kedro-viz/issues/2014

## Description
We use z-index everywhere and just add random number that works. It has become cumbersome to debug or fix any styling as z-indexes being added on top of each other.

## Development notes

Introduced specific z-index variables in `variables.scss` to replace hard-coded values throughout the codebase.

Introduced `$base`, `$above`, and `$below` variables to create a flexible system for defining z-index values. This method provides a clear and scalable way to layer elements relative to each other, avoiding conflicts and magic numbers:

`$base`: 0;
`$above`: 1;
`$below`: -1;

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/styles/_variables.scss [hunk 1]
```diff
@@ -27,10 +27,67 @@ $metadata-modal-breakpoint: 700px;
...
+$zindex-metadata-panel: $base + ($above * 3);
+$zindex-metadata-link: $base + $above;
+
+$zindex-nodelist-placeholder: $base + ($above * 2);
+$zindex-nodelist-heading: $base + $above;
+
+$zindex-group-background-fade: $base + $below;
+
+$zindex-switch-root: $base;
+$zindex-switch-base: $base + $above;
```
Comment (Huongg) [inline src/styles/_variables.scss:71]: hey this is a good start, i think the idea is to set something as a base, in this case 
`$zindex-switch-root: $base;`

then for `$zindex-switch-base: $above + $zindex-switch-root`
and `$zindex-switch-track: $below + $zindex-switch-root`

File: src/styles/_variables.scss [hunk 2]
```diff
@@ -27,11 +27,44 @@ $metadata-modal-breakpoint: 700px;
 $global-toolbar-width: 80px;
 $run-list-controls-height: 95px;
 
-// -- levels -- //
-$z-index-metadata-panel: 3;
-$z-index-metadata-code: 3;
-$z-index-MuiTreeItem-icon: 1;
-$z-index-banner: 6;
+// z-index values
```
Comment (rashidakanchwala) [inline src/styles/_variables.scss:30]: This is nice, can we organize it by levels by grouping all with the at the same level?

**Comment by jitu5** [comment]
@SajidAlamQB 
In this PR we are removing hardcoded z-index values and moved to `_variables.scss` file, But we are not using `$base: 0; $above: 1; $below: -1;` concept right ?

**Review by rashidakanchwala — APPROVED** [review]
Look's good to me :) Much clearer now . thanks

**Comment by SajidAlamQB** [comment]
> @SajidAlamQB In this PR we are removing hardcoded z-index values and moved to `_variables.scss` file, But we are not using `$base: 0; $above: 1; $below: -1;` concept right ?

Yes, that is correct.

**Review by jitu5 — APPROVED** [review]
Thanks @SajidAlamQB.

**Pull request merged**